### ISSUE TITLE
Inflectional classes: which labels?

### ISSUE THREAD

**Issue by Stormur** [UniversalDependencies/docs#748]
I'm probably preceeding @dan-zeman in the quest for standardising features, and hope not to duplicate older issues (I couldn't find others by using the _inflection_ keyword).

So, I was wondering which (language-specific) features could be used to mark **inflectional classes** of words, and if there exists any already.

It is true that such information may not be purely morphological, but rather lexical, as it is often arbitrary from a synchronic point of view: it is not necessarily dependent on a given part of speech nor gender, even if, at the same time, there exist more or less strong correlations. It is this orthogonality that would motivate such feature, and not secondarily the fact that we often find this information in the Latin treebanks we are taking care of, and it would be a pity to lose it during conversion into the UD standard. 

We were experimenting with the already existing [NounClass](https://universaldependencies.org/u/feat/NounClass.html), envisioning a corresponding VerbClass (not yet attested), in parallel to [Variant](https://universaldependencies.org/be/feat/Variant.html) or [Form](https://universaldependencies.org/ga/feat/Form.html), both already existing, mainly for Czech and Irish.

* For noun classes, we were trying to define values that can (hopefully) be applied to any Indo-European language, if applicable, inspired by Bantu and Wolof classes. So, an `IndEurO` value, applied e.g. to Latin _lup**us**_ (_lupi_, _lupo_, _luporum_...) 'wolf' might as well apply to Greek _άνθρωπ**ο**ς_, Lithuanian _mišk**as**_ (please, IEists correct me if I am wrong), and so on. This would be very nice and desirable for interlinguistic comparisons! So we would have  `IndEurA` (_rosa_), `IndEurE` (_spes_), `IndEurI` (_possibilis_), `IndEurO` (_lupus_), `IndEurU` (_domus_), `IndEurX` (i.e. athematic, _rex_ = _reg_ + _s_), alongside the universal "non-value" `Ind` for truly indeclinable (_fas_, _tot_).
    * Now, sometimes a form follows a Greek-like ending of the same class rather than the expected Latin one, e.g. _systema_ 'system' may appear as _systemat**os**_ in the genitive rather than _systemat**is**_. It is often a question of style (= showing off your Greek). We have identified `Variant=Greek` for this, but maybe also `Form=Greek`. The `Foreign` feature does not accomplish the same thing here, since these are words which are fully settled in Latin, and we want to highlight a specific paradigmatic variance. 
* For verb classes, values are probably language-specific, so we define `LatA` for _amo_, _am**a**re_ 'to love', and so on, very similarly than for nouns.
* The two types of inflections need to be kept separated, because it happens that nominal verbal forms, as their name implies, can show both, e.g. _am**a**t**us**_ 'loved' from _amo_ is at the same time a nominal second declension (_o_-theme), and a verbal first conjugation (_a_-theme), so, in our terms, `IndEurO` & `LatA`. 

One thing that brought us towards the for now Bantu-only `NounClass` is that, from its description: 
1) we are in the same realm of lexical properties expressed through morphology; 
2) a language like Latin already has the independent `Gender` feature; 
3) there is a synchronic unpredictability (as explained for Wolof). 

What sets Latin classes apart from Bantu ones is that they are not related to concordance phenomena... I don't know how much this is relevant. I have to admit, I am still a little bit confused about the distinction between `NounClass` and `Gender`, but if they are indeed considered to be separate phenomena, our first concern was to reuse something that was already there. On the contrary, if they aren't, how much possible is it to let `Gender` become just a subtype of `NounClass`?

Other considered features: `NounType` and `VerbType` are already taken and are meant for different things; there is a mysterious `Uninflect` for _uk_ (Ukrainian?). 

I am curious to know what the UD community thinks about this, if there are any suggestions or proposals, and if some treebank has already dealt with such issues! :slightly_smiling_face:

**Comment by dan-zeman** [comment]
The noun classes in Niger-Congo languages, as I understand them, are something else than inflectional classes in Indo-European languages. While they do correspond to nominal inflection at least in Bantu, they are also a property of nouns that can be cross-referenced by other words, most notably verbs. That makes them important in the language system, beyond morphology.

BTW we have copied the values `Bantu1-Bantu23` from the specification of UniMorph but they have not been used in a UD treebank yet (we still don't have Bantu languages). In contrast, there are also 12 classes for Wolof (`Wol1-Wol12`), and these have been applied to real data.

**Comment by Stormur** [comment]
OK. This is somewhat unfortunate, because _class_ is really the right generic term for inflectional paradigms.  

Do you have other naming suggestions? What do you think of:
* `Declension` for nouns, of course keeping the proposed Indo-Europeans values + `Ind` (or maybe a variant of `Uninflecting` to avoid overlaps; but this label is intended for those words that, contrary to the prototypical behaviour of their class, do not show any morphological variations)
* `Conjugation` for verbs

Neither yet exists at the moment.

**Comment by dan-zeman** [comment]
Why not simply `InflClass`?

But I'm a bit skeptical on maintaining one set of values across a family. Perhaps it would be enough to keep the name of the feature cross-linguistically, while the values would always be language-specific. Anyways, we have yet to see how many other people will actually want to use it. For instance, the feature would be perfectly relevant for Czech but I don't have the data in the original annotation and I'm not going to try to obtain it.

**Comment by Stormur** [comment]
> Why not simply `InflClass`?

There should be a distinction between verbal and nominal inflections.

> But I'm a bit skeptical on maintaining one set of values across a family. Perhaps it would be enough to keep the name of the feature cross-linguistically, while the values would always be language-specific. Anyways, we have yet to see how many other people will actually want to use it. For instance, the feature would be perfectly relevant for Czech but I don't have the data in the original annotation and I'm not going to try to obtain it.

I think it is better to open it to the widest possible applications! :slightly_smiling_face: We have this difficulty for some Latin treebanks, too, but I feel that at least the potential of having such a feature is a good thing.

PS: I think the _universal_ label is still valid!

**Comment by dan-zeman** [comment]
> > Why not simply `InflClass`?

> There should be a distinction between verbal and nominal inflections.

Why? Do you intend to combine both on one word? And what would you do with inflections that are neither nominal nor verbal? If the distinction is necessary, it is also possible to start values of nominal inflection with "N" and verbal with "V".

**Comment by Stormur** [comment]
Yes, as I explain in the first post: inflectional classes might stack! In Latin, this happens for participles:
* Perfect participle: _am**a**re_ `VerbClass=LatA` -> _am**a**t**us**_ `NounClass=IndEurO` (_amati_, _amatorum_...)
* Present participle: _aud**i**re_ `VerbClass=LatI` -> _aud**ie**n**s**_ `NounClass=IndEurI` (_audientis_, _audientium_...)

The fact is that both pieces of information are relevant from an inflectional point of view.

I can imagine that other kinds of stackings can happen in ways that I cannot fathom, of course not limited to verbal conjugation + nominal declension. Probably, also multiple nominal inflectional paradigms can happen at the same time. This was my original motivation for having two features. But it has occurred to me that we may solve this, allowing for as great as possible flexibility, the following way, starting from your proposals:

* the basic feature is `InflClass`;
* stackings use layers. Their definition seems perfectly fitting here. So, for _amatus_ above, we would have `InflClass[verb]=LatA|InflClass[noun]=IndEurO` (where `noun` is used as a label for _nominal_ in general). 
* This means that inflectional paradigms need not necessarily be distinguished at the value level, even if they might (e.g. `NIndEurO` for "nominal declensions with _o_ theme", `VLatA` for "verbal Latin conjugation with a theme"), because the combination of UPOS and `InflClass` is already enough.
